src/web-ui/src/App.js

```diff
@@ -68,119 +58,119 @@ class App extends Component {
         gateway.postEngagement(result).then(() => {});
         return result;
       });
-
-      this.setState({ detectedFaces }, () => {
-        if (this.state.rekognizing) {
-          setTimeout(this.getSnapshot, 300);
-        }
-      });
+      setDetectedFaces(detectedFaces);
+      setIterationCompleted(true);
     });
 
     gateway.searchFaces(b64Encoded).then(response => {
       const detectedPeople = [];
       if (response.FaceMatches) {
         response.FaceMatches.forEach(match => {
           const externalImageId = match.Face.ExternalImageId;
-          detectedPeople.push(
-            this.state.people.find(x => x.externalImageId === externalImageId)
+          const matched = people.find(
+            x => x.externalImageId === externalImageId
           );
+          if (matched) detectedPeople.push(matched);
         });
       }
-      this.setState({ detectedPeople });
+      setDetectedPeople(detectedPeople);
     });
-  }
-
-  toggleRekognition() {
-    this.setState(
-      {
-        rekognizing: !this.state.rekognizing
-      },
-      () => {
-        if (this.state.rekognizing) {
-          gateway.getPeople().then(response => {
-            this.setState({ people: response.people }, this.getSnapshot);
-          });
-        }
-      }
-    );
-  }
+  };
+
+  const setupWebcam = instance => {
```

Comment:
This looks weird to me, not sure why you want to set a state inside the ref, also I think you can switch to use useRef

Reply:
Yes, you were right. I was very confused about useRef, at the end that's exactly what I needed, also to handle the toggling mechanism